tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/AdminFunctionalityTests.cs

```diff
@@ -116,6 +116,32 @@ public async void TestAddSanitizer()
             Assert.True(result is HeaderRegexSanitizer);
         }
 
+        [Fact]
+        public async void TestAddSanitizerWithOddDefaults()
+        {
+            // arrange
+            RecordingHandler testRecordingHandler = new RecordingHandler(Directory.GetCurrentDirectory());
+            var httpContext = new DefaultHttpContext();
+
+            httpContext.Request.Headers["x-abstraction-identifier"] = "BodyKeySanitizer";
+            httpContext.Request.Body = TestHelpers.GenerateStreamRequestBody("{ \"jsonPath\": \"$.TableName\", \"value\": \"Sanitized\" }");
+            httpContext.Request.Headers["Content-Length"] = new string[] { "60" };
+            httpContext.Request.Headers["Content-Type"] = new string[] { "application/json" };
+
+            var controller = new Admin(testRecordingHandler)
+            {
+                ControllerContext = new ControllerContext()
+                {
+                    HttpContext = httpContext
+                }
+            };
+            testRecordingHandler.Sanitizers.Clear();
+            await controller.AddSanitizer();
+
+            var result = testRecordingHandler.Sanitizers.First();
+            Assert.True(result is BodyKeySanitizer);
```

Comment:
Can we add an assertion on the "value" being used?

Reply:
Amendment, not in this PR. I need to update the `sanitizers` to have public members. I can't see into them currently. I should be able flip those public without must affect to testing/features (info/active may need some work), but that's a bit bigger than a change than I want to go after in this PR. I've also got bunch of higher pri investigations that I need to get a move on in the near future.

https://github.com/Azure/azure-sdk-tools/issues/2304

Reply:
One thing you can do is add the InternalsVisibleTo attribute in the assemblyInfo file for the TestProxy, and specify your test project - e.g. https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventgrid/Azure.Messaging.EventGrid/src/Properties/AssemblyInfo.cs#L6

Reply:
That's super interesting! I will make note of this possibility 👍 .

I don't think it's worth going their complicated, you're right in that It's not like they NEED to be private members.